### ISSUE TITLE
Function on attrs object mistaken for bound method

### ISSUE THREAD

**Issue by meshy** [PyCQA/pylint#3801]
### Steps to reproduce
Run pylint on the following:

```python
from typing import Callable
​
import attr
​
​
def function(value: int) -> None:
    print(value)
​
​
@attr.s(frozen=True)
class A:
    value: int
    outsourced: Callable = function
​
    def execute(self) -> None:
        self.outsourced(self.value)  # E1121 is here
​
​
a = A(1)
a.execute()
```

### Current behavior

`E1121: Too many positional arguments for function call (too-many-function-args)`

### Expected behavior

No error. I believe `self.outsourced` is being mistaken for a bound method. Running the code proves that it is not.

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 27 2020, 08:42:51) 
[GCC 10.1.0]
```

**Comment by meshy** [comment]
Interestingly, when I remove `function` as a default argument to `outsourced`, the error goes away.

**Comment by hippo91** [comment]
@meshy thanks for the report.
I can reproduce the bug with your code slightly modified: https://gist.github.com/hippo91/13c5813cf100a59cd32253f0caf9ff1e

python3.7
pylint master branch
astroid 1.4

**Comment by meshy** [comment]
Apologies for the slight discrepancy! Thanks for looking into this :)
